packages/connection-form/src/components/connection-form.tsx

```diff
@@ -429,16 +422,58 @@ function ConnectionForm({
     },
     [onSaveClicked, setErrors]
   );
+  const onSubmitForm = useCallback(() => {
+    const updatedConnectionOptions = cloneDeep(connectionOptions);
+    // TODO: this method throws on malformed connection strings instead of
+    // returning errors
+    const formErrors = validateConnectionOptionsErrors(
+      updatedConnectionOptions
+    );
+    if (formErrors.length) {
+      setErrors(formErrors);
+      return;
+    }
+    if (disableEditingConnectedConnection) {
+      // there isn't a connect button in this case, so the default action should
+      // be to save, not to save and connect
+      void callOnSaveConnectionClickedAndStoreErrors?.(
+        getConnectionInfoToSave()
+      );
+    } else {
+      onSaveAndConnectClicked?.({
+        ...initialConnectionInfo,
+        ...getConnectionInfoToSave(),
+        connectionOptions: updatedConnectionOptions,
+      });
+    }
+  }, [
+    connectionOptions,
+    disableEditingConnectedConnection,
+    setErrors,
+    callOnSaveConnectionClickedAndStoreErrors,
+    getConnectionInfoToSave,
+    onSaveAndConnectClicked,
+    initialConnectionInfo,
+  ]);
 
   const showPersonalisationForm = useConnectionFormSetting(
     'showPersonalisationForm'
   );
 
   const showHelpCardsInForm = useConnectionFormSetting('showHelpCardsInForm');
 
+  /*
+  To test onSubmit on a form (ie. what happens when you press {enter} and not
+  what happens when you click a submit button) you have to use submit() which only
+  works with role="form", but eslint doesn't like that because role="form"
+  should be redundant.
+  https://stackoverflow.com/questions/59362804/pressing-enter-to-submit-form-in-react-testing-library-does-not-work
```

Comment:
If there's a better way to simulate submitting a form by pressing enter rather than pressing a button so that the callback gets called in tests, please let me know.